### PR TITLE
Adding java support

### PR DESCRIPTION
**Pull request by legeriergeek**

### CODE REVIEW

**Comment by Symmettry** [comment]
damn i was just about to finish my pr for this

**Comment by Symmettry** [comment]
oh ok there's quite a few flaws in your code anyway.

firstly you have two java.lua

secondly you stole the js comments (most of them don't even apply)

third your function and variable detection doesn't work

fourth you're missing a lot of highlights and are highlighting a lot of things that don't need to be highlighted

**Comment by Symmettry** [comment]
java does not have "in" or "of"
this is not javascript

File: Lua/Plugins/java.lua [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+add_comment("mf the typa guy to trigger \"require is not defined\"")
+add_comment("im scared")
+add_comment("more?.optional?.chaining?.please // LGTM")
+add_comment("tip: remove runtime errors by removing null: rm -rf /dev/null")
+add_comment("JAVASCRIPTTTTTTTTTTTT LETS FUCKING GOOOO ==== X ?? Y ? GOOOO : X 🗣️🗣️🗣️")
+add_comment("wrap it in do {} while(false) 👍👍👍")
+add_comment("\"undefined\" is not defined.")
+add_comment("make your JS roleplay Java, change that i++ to i--")
+add_comment("tip: decrease your chance at errors: Number.MAX_SAFE_INTEGER += Number.MAX_SAFE_INTEGER")
+add_comment("tip: Use VSCODE")
```
Comment (RealPacket) [inline Lua/Plugins/java.lua:103]: that editor sucks

File: Lua/Plugins/java.lua [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+highlight("{", "binary")
+highlight("}", "binary")
+highlight("[", "binary")
+highlight("]", "binary")
+
+
+
+highlight_region("\"", "\"", "string")
+highlight_region("'", "'", "string")
+highlight_region("`", "`", "string")
```
Comment (RealPacket) [inline Lua/Plugins/java.lua:79]:
```suggestion
```

**Comment by Symmettry** [comment]
You can't even do ' in java.
See #32 where I implemented java without all these huge mistakes

**Comment by RealPacket** [comment]
> You can't even do ' in java. See #32 where I implemented java without all these huge mistakes

You can, except that it's a character instead of a string.

**Comment by Symmettry** [comment]
Ah true. I rarely do it so that's why I forgot it. Kinda annoying to try to implement with the current version of griddycode as it's very limited (especially in the color department)

**Comment by face-hh** [comment]
Implemented in #32

File: Lua/Plugins/java.lua (hunk 1 above)
Comment (elefantapan) [reply Lua/Plugins/java.lua:103]: True

**Pull request closed without merging**